### ISSUE TITLE
Importing an entry with the browser extension doesn't delete temporary files

### ISSUE THREAD

**Issue by ror3d** [JabRef/jabref#11048]
### JabRef version

5.12 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 11

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Edit the JabRefHost.ps1 file in your JabRef installation
2. Change line 40 (`Remove-Item...`) with the following:
   ````
    $rout = Remove-Item $tempfile *>&1
    $wshell = New-Object -ComObject Wscript.Shell
    $wshell.Popup("ROut: $rout", 0, "JabRef", 0x0 + 0x30)
   ````
4. Open JabRef
5. Use the JabRef browser extension in any article to try to import it
6. Upon import, a popup window will open showing the output of the `Remove-Item` command, reporting that it cannot remove the temporary file because it is being used:
   ![image](https://github.com/JabRef/jabref/assets/1844478/4bb2f40b-3792-4376-9be3-eb9cdbf20fef)

This is probably due to the JabRef command being called unblocking. The script has to exit to send back the result to the browser, but the open JabRef instance might still have the file open, so the script can't delete it before exiting.

Being able to send the file as stdin, or as a command-line argument, instead of creating a temporary file, could be a solution for this issue. Another approach could be connection using sockets in some way.

**Comment by ror3d** [comment]
This is related to https://github.com/JabRef/JabRef-Browser-Extension/issues/493 but that one is about the issue of the extension reporting an error, which should be solved with PR #11047, whereas this issue is about the still unresolved problem left in that PR.

**Comment by tobiasdiez** [comment]
> Being able to send the file as stdin, or as a command-line argument, instead of creating a temporary file, could be a solution for this issue. 

This is what we had before. But some of the bibtex can be quite long and then you hit the maximum length of command line argument. 

Is it really important to you to delete the temporary file? Windows 10+ normally clean the temp folder automatically, so it shouldn't really be a problem that there are still files lying around. I honestly don't see a clean way to clean them up, since as you say we need to return as fast as possible from the script.

It would perhaps work to pass a `--deleteWhenClosing` cmd line arg to jabref's main exe, and then clean up the file when the main jabref program is closed - but that feels a bit overkill.

**Comment by ror3d** [comment]
Right, as command-line argument would have such problem, sending them through stdin should work well though.

On the other hand, I am also thinking of ways that https://github.com/JabRef/JabRef-Browser-Extension/issues/481 could use a similar approach, but that might be a bit more tricky.

**Comment by Deehuh** [comment]
Hi, I'm working with a group of 2 other CS students from Florida International University. We are part of the Tech Talent Academy program where we are required to join an open source project and make a contribution. We were wondering if we could tackle this issue

**Comment by ThiloteE** [comment]
@Deehuh since you and your teammates already have been assigned to another issue, I will not assign you to this one to avoid confusion. I expect you will only work on one issue anway, right?

**Comment by Deehuh** [comment]
Yes, that's correct. Thank you.

**Comment by Noah-Martin1** [comment]
Hello, has anyone been assigned this issue yet? I am from ANU in australia and would like to work on this issue with my group for an assignment.

**Comment by Noah-Martin1** [comment]
I apologize for the inconvenience, but I’ve been working on this issue and unfortunately, my Mac doesn’t have sufficient RAM to run both VirtualBox and IntelliJ simultaneously. Given that this issue pertains to Windows, it’s quite challenging to resolve it on my current setup.

Please unsassign me from this issue.

**Comment by koppor** [comment]
> I apologize for the inconvenience, but I’ve been working on this issue and unfortunately, my Mac doesn’t have sufficient RAM to run both VirtualBox and IntelliJ simultaneously. 

On any chance, did you try whether this issue also happens on MacOS setup?

**Comment by koppor** [comment]
> Windows 10+ normally clean the temp folder automatically,

It does it when running tghe "Disk Cleanup tool" with a nice configuration - source: https://superuser.com/a/902645/138868

**Comment by Noah-Martin1** [comment]
> > I apologize for the inconvenience, but I’ve been working on this issue and unfortunately, my Mac doesn’t have sufficient RAM to run both VirtualBox and IntelliJ simultaneously.
> 
> On any chance, did you try whether this issue also happens on MacOS setup?

No, I didn't extensively check if this issue happens on MacOS.

**Comment by pranav0510s** [comment]
Hello I am a student of Uni Adelaide and i wanted to get assigned to this issue for my Project at my Uni

**Comment by koppor** [comment]
> Hello I am a student of Uni Adelaide and i wanted to get assigned to this issue for my Project at my Uni 

👋 Hey @pranav0510s, looks like you’re eager to work on this issue—great! 🎉 It also looks like you skipped reading our [CONTRIBUTING.md](https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md), which explains exactly how to participate. No worries, it happens to the best of us. Give it a read, and you’ll discover the ancient wisdom of assigning issues to yourself. Trust me, it’s worth it. 🚀

**Comment by koppor** [comment]
@pranav0510s Here I have a line break at the help. Did that happen to you, too?

![Screenshot_20250501-192002.png](https://github.com/user-attachments/assets/c3304f2f-3887-41b9-a4af-bc38f4e67ed8)

**Comment by Andrija121** [comment]
Hi I am very interested in trying to solve this issue. 
Would it be ok for it to be assigned to me?
Although I have not worked on gfi so far, some of my team members which with whom I am in touch already picked a gfi, and I was just wondering if I could jump to this gsi, as we will be working together in some extent?

Since this one looked really interesting to me and used to be gfi before.
Please let me know and thx!

**Comment by koppor** [comment]
@Andrija121 It was a good *second* issue. I think with recent developments, its more hard. Please search for something else. If you are a team, you should check larger issues (and not something taking 10 minutes with AI). See https://github.com/orgs/JabRef/projects/3/views/3 for a list

**Comment by Andrija121** [comment]
@koppor Thank you for your response. 

Alright, I will take a look at the link you've sent.
